src/main/scala/firrtl/transforms/VerilogPrep.scala

```diff
@@ -0,0 +1,138 @@
+            val name = namespace.newTemp
+            val wire = DefWire(NoInfo, name, instPorts.head.tpe)
+            val ref = WRef(wire)
+            for (inst <- instPorts) sourceMap(inst) = ref
+            wire // Replace the attach with new source wire definition
+          case Seq(source) => // One source can be removed
+            assert(!sourceMap.contains(source)) // should have been merged
+            for (inst <- instPorts) sourceMap(inst) = source
+            EmptyStmt
+          case moreThanOne =>
+            attach
+        }
+      case s => s
+    }
+
+    (m map onStmt, sourceMap.toMap)
+  }
+
+  def remove_root(e: Expression): WRef = {
+    def rec(ex: Expression): Expression = {
+      ex match {
+        case ex: WSubField => ex.expr match {
+          case (e: WSubField) => remove_root(e)
+          case (_: WRef) => WRef(ex.name, ex.tpe, InstanceKind, UNKNOWNGENDER)
+        }
+        case _ => throwInternalError(s"shouldn't be here: remove_root($ex)")
+      }
+    }
+    rec(e) match {
+      case w: WRef => w
+      case _ => throwInternalError(s"shouldn't be here: remove_root($e)")
+    }
+  }
+
+  def execute(cs: CircuitState): CircuitState = {
+    
+    val annosMap = cs.annotations
+      .collect{ case a : SingleTargetAnnotation[_] => a }
+      .groupBy(_.target)
+    
+    var annos = cs.annotations.toSeq.toList
+    val circuitTarget = CircuitTarget(cs.circuit.main)
+    
+    def lowerM(attachMap: AttachSourceMap, moduleName: String, modx: DefModule): DefModule = {
+      val moduleTarget = circuitTarget.module(moduleName)
+        
+      def lowerE(e: Expression): Expression = e match {
+        case (_: WRef | _: WSubField) if kind(e) == InstanceKind =>
+          WRef(LowerTypes.loweredName(e), e.tpe, kind(e), gender(e))
+        case _ => e map lowerE
+      }
+
+      def lowerS(attachMap: AttachSourceMap)(s: Statement): Statement = s match {
+        case WDefInstance(info, name, module, tpe) =>
+          val portRefs = create_exps(WRef(name, tpe, ExpKind, MALE))
+          val (portCons, wires) = portRefs.map { p =>
+            attachMap.get(p) match {
+              // If it has a source in attachMap use that
+              case Some(ref) => (p -> ref, None)
+              // If no source, create a wire corresponding to the port and connect it up
+              case None =>
+                val lname = LowerTypes.loweredName(p)
+                val wire = DefWire(info, lname, p.tpe)
+                val target = moduleTarget.ref(name).field(remove_root(p).name)
+                if(annosMap.contains(target)){
+                  annosMap(target).foreach((anno : SingleTargetAnnotation[_]) => {
+                    anno.target match {
+                      case t: IsMember => 
+                        val r = RenameMap()
+                        r.record(t,Seq(moduleTarget.ref(lname)))
+                        annos ++= anno.update(r)
```

Comment:
see comment in PR : 
current approach is to duplicate any `SingleAnnotationTarget[]` of kind `Circuit|Module>instance.io.reset` into:
- `Circuit|Module>instance.io.reset`
- `Circuit|Module>instance_io_reset`

Hence it cannot be done with a simple RenameMap returned to the compiler

Reply:
Not looking at the logic here, but maybe you're looking for the [`duplicate` method on `SingleTargetAnnotation`]
(https://www.chisel-lang.org/api/firrtl/SNAPSHOT/firrtl/annotations/SingleTargetAnnotation.html#duplicate(n:T):firrtl.annotations.Annotation)?

Reply:
precisely! thanks for pointing it :)
Still don't know if modifying `VerilogPrep` is the good approach though

Reply:
Well actually it is not that simple at all if I want to remain generic on duplicating any kind of annotation (which might not be a good idea?)
https://github.com/freechipsproject/firrtl/blob/master/src/main/scala/firrtl/annotations/Annotation.scala#L51-L78

Reply:
Looking at this again, I think Adam is right. Why can't this be done with a `RenameMap` as that can allow renames to multiple things? One of those renames could be itself, I think.

Reply:
```scala
// beginning
val renameMap = RenameMap()

// core
renameMap.record(t, Seq(t,moduleTarget.ref(lname)))

// end
cs.copy(circuit = cs.circuit.copy(modules = modulesx), renames = Some(renameMap))
```

This works well indeed, but it doesn't answer the question : is this systematic duplication a good idea?
 I am not sure this would be the expected behavior for anyone annotating an instance port : why the associated wire definition and assign should get annotated as well ?
Although it fits well my current case,  it might be a confusing behavior?
On the other hand this duplication is only to be seen by `AddDescriptionNodes` transform and the Emitter itself.